internal-parallel.h

```diff
@@ -184,7 +184,7 @@ static void piter_bisect(struct piter * pi, char * P) {
              * the range is too small, so we set flag narrow, 
              * and next iteration we will directly test Pright */
             if(d->compar(&P[i * d->rsize], 
-                &pi->Pleft[i * d->rsize], d->rsize) == 0) {
+                &pi->Pleft[i * d->rsize], d->rsize) <= 0) {
```

Comment:
This looks like a reasonable cure, but we are not at the bottom of it. Something must have been seriously wrong if after bisect P is not inside [left right].

I am thinking of adding an assertion before and after 180, die and print everything if the precondition that P is in side [left, right] is wrong.

If you have other priorities, I can take a closer look into this over the weekend.